src/OSPSuite.Core/Domain/Formulas/DistributionFormulaFactory.cs

```diff
@@ -21,68 +25,85 @@ public interface IDistributionFormulaFactory
 
-      public DistributionFormulaFactory(IObjectPathFactory objectPathFactory,IObjectBaseFactory objectBaseFactory)
+      public DistributionFormulaFactory(IObjectPathFactory objectPathFactory, IObjectBaseFactory objectBaseFactory)
       {
          _objectPathFactory = objectPathFactory;
          _objectBaseFactory = objectBaseFactory;
       }
 
-      public NormalDistributionFormula CreateNormalDistributionFormulaFor(IDistributedParameter distributedParameter, IParameter meanParameter, IParameter deviationParameter)
+      public NormalDistributionFormula CreateNormalDistributionFormulaFor(IDistributedParameter distributedParameter, IParameter meanParameter, IParameter deviationParameter) =>
+         createDistribution<NormalDistributionFormula>(meanParameter.Dimension, configureNormalDistribution(meanParameter.Name, deviationParameter.Name));
+
+      private Action<DistributionFormula, IDimension> configureNormalDistribution(string meanParameterName = null, string deviationParameterName = null) => (distributionFormula, dimension) =>
       {
-         var distribution = createDistribution<NormalDistributionFormula>();
-         distribution.Dimension = meanParameter.Dimension;
-         addObjectPathToDistributionFormula(distribution, distributedParameter,meanParameter,Constants.Distribution.MEAN);
-         addObjectPathToDistributionFormula(distribution, distributedParameter,deviationParameter,Constants.Distribution.DEVIATION);
-         return distribution;
-      }
+         addObjectPathToDistributionFormula(distributionFormula, meanParameterName, MEAN, dimension);
+         addObjectPathToDistributionFormula(distributionFormula, deviationParameterName, DEVIATION, dimension);
+      };
 
-      public DiscreteDistributionFormula CreateDiscreteDistributionFormulaFor(IDistributedParameter distributedParameter, IParameter meanParameter)
+      public DiscreteDistributionFormula CreateDiscreteDistributionFormulaFor(IDistributedParameter distributedParameter, IParameter meanParameter) =>
+         createDistribution<DiscreteDistributionFormula>(meanParameter.Dimension, configureDiscreteDistribution(meanParameter.Name));
+
+      private Action<DistributionFormula, IDimension> configureDiscreteDistribution(string meanParameterName = null) => (distributionFormula, dimension) =>
+         addObjectPathToDistributionFormula(distributionFormula, meanParameterName, MEAN, dimension);
+
+      public LogNormalDistributionFormula CreateLogNormalDistributionFormulaFor(IDistributedParameter distributedParameter, IParameter meanParameter, IParameter deviationParameter) =>
+         createDistribution<LogNormalDistributionFormula>(meanParameter.Dimension, configureLogNormalDistributionDistribution(meanParameter.Name, deviationParameter.Name));
+
+      private Action<DistributionFormula, IDimension> configureLogNormalDistributionDistribution(string meanParameterName = null, string deviationParameter = null) => (distributionFormula, dimension) =>
```

Comment:
same

Reply:
rename **configureLogNormalDistributionDistribution** to **configureLogNormalDistribution** :)